# variant-caller

<h1> Repository Contents: </h1>
 
<h4> 	Simulated Mutant Genome: Validation code python file </h4>
<p>Test files</p>
<ul>   
  <li> EcoliK12-MG1655.fasta</li>
  <li> NC320767.1.fasta </li>
 </ul>
<p> How to use: </p>
<ul> <li>Download dependencies (files and conda environment (samtools, bcftools, snippy)</li>
•	<li> Change to files directory </li>
<li> Run python file from command line with three input variables </li>
  <ol> Mandatory: Reference fasta to be mutated </ol>
  <ol> Mandatory: Output VCF file name (truth) </ol>
  <ol> -o output fasta file name </ol>
</ul>

<ul>
<h4> Bash script for simulating 100bp reads at 30x depth </h4>	
<p>How to use: </p>
<li> Open terminal and run with three input variables </li>
<li> Calculate number of reads required for simulating 30x coverage </li>
<li> Runs Wgsim (make sure “conda install samtools::wgsim”) </li>	
<li> Outputs two fastq files into your current directory </li>

<h4> Bash - Pipeline (reference-based) </h4>
<p> How to use: </p>
<ul> 
<li> Receive fasta, fastq files (in total 3 variables) </li>
<li> Output bam file for vcf calling (snippy and Bcftools per genome) </li>
</ul>

<h4> Bash - Merge VCFs (bcftools merge) </h4>
<ul> 
<li> Run from terminal with names of both VCF files to merge (make sure they are in the current directory as snippy saves output to new folder) </li>
</ul>

<h4> Discussion on performance </h4>
<p> The truth VCF for mutant genomes are generated by the validation python code. It can be used to assess variant caller performance. Important parameters to measure performance are the specificity (TPs) and sensitivity (TP/FP) </p>
<p> Errors with alignment stage using Snippy pipeline: samtools view 'filter'  </p>
